dracut/modules.d/99kiwi-lib/kiwi-filesystem-lib.sh

```diff
@@ -4,19 +4,34 @@ function resize_filesystem {
     local device=$1
     test -n "${device}" || return
     local resize_fs
+    local set_label="true"
     local mpoint=/fs-resize
     local fstype
+    local fslabel
+    local fslabel_final
     fstype=$(probe_filesystem "${device}")
+    fslabel=$(blkid "${device}" -s LABEL -o value)
+    if [[ "${fslabel}" == *"_IMG" ]];then
+        # The filesystem label is the one set from the image build
+        # this should indicate this is the first time this image
+        # got deployed on the target. Along with the resize of
+        # the filesystem that label will now change to its final
+        # state
+        fslabel_final=${fslabel%%_*}
```

Comment:
this is wrong. If the label is FOO_BAR, it gets FOO_BAR_IMG. But now you change it to FOO.

Better:
```
   fslabel_final=${fslabel%_*}
```
best:
```
   fslabel_final=${fslabel%_IMG}
```

OK, I see that only A-Za-z are allowed as label-name (*which is a backwards-compatibility breaking change!*), but still the code should be conservative.

Reply:
I see you already realized the connected schema change which only allows a-zA-Z labels. So the situation you constructed is correct but could not be created with kiwi. Of course there is an opportunity for the user to change the label as he wants but that could also be used to mess with the entire idea. I'm not sure if it makes sense to care for your side effect if there are still other side effects we can't control. In theory a user can set a new label as part of the config.sh script and invalidate the entire mechanism. That's the reason why I kept the code as simple as possible even though I admit it's only simple together with the restrictions on the label in the schema.

Reply:
> (which is a backwards-compatibility breaking change!)

Not necessarily. Before there was no idea for compatibility. We allowed { text } for the filesystem label. But every label of a filesystem has certain constraints, 11-16 characters lengths only a subset of symbols allowed. The constrain I set now 8 characters and only a-zA-Z is a safe label name across all kiwi supported filesystems. So this change is not backward imcompatible but protecting users from wrong label definitions :)